base_classes/NXidentifier.nxdl.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<?xml-stylesheet type="text/xsl" href="nxdlformat.xsl"?>
+<!--
+# NeXus - Neutron and X-ray Common Data Format
+#
+# Copyright (C) 2014-2024 NeXus International Advisory Committee (NIAC)
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXidentifier" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         An identifier for a (persistent) resource, e.g., a DOI or orcid.
+    </doc>
+    <field name="service" type="NX_CHAR">
+        <doc>
+             The service by which the resource can be resolved.
+             
+             Examples: doi, urn, hdl, purl, orcid, iso, url
+        </doc>
+    </field>
+    <field name="identifier" type="NX_CHAR">
+        <doc>
+             The unique code, IRI or hash to resolve this reference.
+             Typically, this is stated by the service which is considered a complete
+             identifier, e.g., for a DOI it's something of the form `10.1107/S1600576714027575`
+             or `https://doi.org/10.1107/S1600576714027575`, which are both resolvable.
+        </doc>
+    </field>
+    <field name="is_persistent" type="NX_BOOLEAN">
```

Comment:
I'm not sure what is the use-case for this field.

For many types of PIDs, the persistency is tacit: an ORCID is a persistent identifier, so why specify this?  Put another way, what does it mean to specify an NXidentifier with `service` "ORCID" and `is_persistent` of "false"?

I suppose the intention here is to allow ad-hoc values in "service" field and allow software to discover these NXidentifier entries are actually PIDs.  Perhaps, to allow software to discover that DOIs are persistent identifiers without that software knowing what are DOIs.

However, in most cases, persistency is a feature of the underlying identifier "service" (or "type").  Including this field seems like it is inviting inconsistent entries.